core/src/main/scala/org/locationtech/rasterframes/expressions/localops/Clip.scala

```diff
@@ -0,0 +1,74 @@
+package org.locationtech.rasterframes.expressions.localops
+
+import org.apache.spark.sql.Column
+import org.apache.spark.sql.catalyst.analysis.TypeCheckResult
+import org.apache.spark.sql.catalyst.analysis.TypeCheckResult.{TypeCheckFailure, TypeCheckSuccess}
+import org.apache.spark.sql.catalyst.expressions.codegen.CodegenFallback
+import org.apache.spark.sql.catalyst.expressions.{Expression, ExpressionDescription, TernaryExpression}
+import org.apache.spark.sql.functions.lit
+import org.apache.spark.sql.rf.TileUDT
+import org.apache.spark.sql.types.DataType
+import org.locationtech.rasterframes.encoders.CatalystSerializer._
+import org.locationtech.rasterframes.expressions.DynamicExtractors._
+import org.locationtech.rasterframes.expressions.row
+
+@ExpressionDescription(
+  usage = "_FUNC_(tile, min, max) - Return the tile with its values clipped to a range defined by min and max," +
+            " doing so cellwise if min or max are tile type",
+  arguments = """
+      Arguments:
+        * tile - the tile to operate on
+        * min - scalar or tile setting the minimum value for each cell
+        * max - scalar or tile setting the maximum value for each cell"""
+)
+case class Clip(left: Expression, middle: Expression, right: Expression)
```

Comment:
There's already precident for calling this kind of operation "clamp". 
https://github.com/locationtech/rasterframes/search?q=clamp&unscoped_q=clamp

A clip in imagery usually implies trimming the pixel edges, where as a clamp is on the pixel values.
https://imagemagick.org/script/command-line-options.php#clamp

So my preference here (sorry) is to keep with that nomenclature.